src/components/side_panel/chart/building_blocks/generic_side_panel/config_panel.ts

```diff
@@ -104,7 +104,26 @@ export class GenericChartConfigPanel extends Component<Props, SpreadsheetChildEn
    * button "confirm" is clicked
    */
   onDataSeriesRangesChanged(ranges: string[]) {
+    let useLeftAxis = false,
+      useRightAxis = false,
+      undefinedAxis = false;
+    for (const design of this.dataSeriesRanges || []) {
+      if (design.yAxisId === "y1") {
+        useRightAxis = true;
+      } else if (design.yAxisId === "y") {
+        useLeftAxis = true;
+      } else {
+        undefinedAxis = true;
+      }
+    }
+    let yAxisId: string | undefined = undefined;
+    if (useLeftAxis && !useRightAxis && !undefinedAxis) {
+      yAxisId = "y";
+    } else if (!useLeftAxis && useRightAxis && !undefinedAxis) {
+      yAxisId = "y1";
```

Comment:
```suggestion
    let yAxisId: string | undefined = undefined;
    if (this.dataSeriesRanges.every(/* hasLeftAxis */) {
      yAxisId = "y";
    } else if (this.dataSeriesRanges.every(/* hasRightAxis */)) {
      yAxisId = "y1";
```
And the above for loop can be removed ? Unless I understand the function worng, it's kinda hard to read ^^'